pyqtgraph/functions.py

```diff
@@ -1752,51 +1747,124 @@ def arrayToQPath(x, y, connect='all'):
-        arr[1:-1]['c'][1::2] = 1  # connect every 2nd point to every 1st one
-    elif eq(connect, 'finite'):
+    if connect == 'all':
+        path.addPolygon(backstore)
+        return path
+    elif connect == 'pairs':
+        if qt6:
+            if n % 2 == 1:
+                arr = arr[:-1]
+            polygon = create_qpolygonf(n // 2 * 3)
+            dst = ndarray_from_qpolygonf(polygon).reshape((-1, 6))   # x0, y0, x1, y1, nan, nan
+            dst[:, :4] = np.frombuffer(arr, dtype=np.double).reshape((-1, 4))   # x0, y0, x1, y1
+            dst[:, 4:] = np.nan
+            path.addPolygon(polygon)
+            return path
+        else:
+            arr['c'][::2] = 0
+            arr['c'][1::2] = 1  # connect every 2nd point to every 1st one
+    elif connect == 'finite':
         # Let's call a point with either x or y being nan is an invalid point.
         # A point will anyway not connect to an invalid point regardless of the
         # 'c' value of the invalid point. Therefore, we should set 'c' to 0 for
         # the next point of an invalid point.
-        if isfinite is None:
+        if qt6:
+            # look for the first finite element.
             isfinite = np.isfinite(x) & np.isfinite(y)
-        arr[2:]['c'] = isfinite
-    elif isinstance(connect, np.ndarray):
-        arr[2:-1]['c'] = connect[:-1]
+            argNaNs = np.argwhere(isfinite)
+            first = 0 if len(argNaNs) == 0 else argNaNs.item(0)
+
+            # if found, set first element to first found finite element
+            arr[0] = arr[first]
+            path.addPolygon(backstore)
+            return path
+        else:
+            if isfinite is None:
+                isfinite = np.isfinite(x) & np.isfinite(y)
+            arr[1:]['c'] = isfinite[:-1]
+    elif connect == 'array':
+        arr[1:]['c'] = connect_array[:-1]
     else:
         raise Exception('connect argument must be "all", "pairs", "finite", or array')
 
-    arr[1]['c'] = 0  # the first vertex has no previous vertex to connect
-
-    byteview.data[-20:-16] = struct.pack('>i', 0)  # cStart
-    byteview.data[-16:-12] = struct.pack('>i', 0)  # fillRule (Qt.OddEvenFill)
-
-    # create datastream object and stream into path
-
-    ## Avoiding this method because QByteArray(str) leaks memory in PySide
-    #buf = QtCore.QByteArray(arr.data[12:lastInd+4])  # I think one unnecessary copy happens here
-
-    path.strn = byteview.data[16:-12]  # make sure data doesn't run away
-    try:
-        buf = QtCore.QByteArray.fromRawData(path.strn)
-    except TypeError:
-        buf = QtCore.QByteArray(bytes(path.strn))
-    except AttributeError:
-        # PyQt6 raises AttributeError
-        buf = QtCore.QByteArray(path.strn, path.strn.nbytes)
+    arr[0]['c'] = 0  # the first vertex has no previous vertex to connect
 
+    # create QDataStream object and stream into QPainterPath
+    path.strn = backstore
+    if QT_LIB == "PyQt6" and qtver < [6, 1, 1]:
```

Comment:
So for the PyQt6 6.1.1-dev wheels, it's QtCore.QT_VERSION == 0x60100 and QtCore.PYQT_VERSION == 0x60101.
So version testing should be:
```python
if QT_LIB == "PyQt6" and QtCore.PYQT_VERSION < 0x60101:
```
We also use PYQT_VERSION here : https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/functions.py#L1406

Reply:
I have no problem with this change, but can you describe the motivation behind using the hexadecimal values instead?

Reply:
oh re-read I assume that using the hex values, it should handle differences in `-dev` wheels?

Reply:
PyQt version does not always match the Qt version. Sometimes PyQt fixes bugs in the bindings and releases a new version without waiting for the corresponding Qt release.

This is the case here, although only for a testing -dev wheel.

Reply:
oh 🤦🏻  yes, difference between library version and Qt version built again, ...yes, I've been bitten by that.  I'll make the change shortly.  I've been poking people to review #1807, I'd feel better getting the blessing from someone else before I merge my own PR of that size.